src/index.ts

```diff
@@ -146,6 +146,29 @@ const App = async () => {
     }
   }
 
+  const populateInputFromURL = () => {
+    const params = new URLSearchParams(window.location.search)
+    const peerOrMaddr = params.get('peer-or-maddr')
+    if (peerOrMaddr) {
+      DOM.input().value = peerOrMaddr
+    }
+  }
+
+  const updateQueryParam = (value: string) => {
+    const url = new URL(window.location.href)
+    if (value) {
+      url.searchParams.set('peer-or-maddr', value)
```

Comment:
may not be necessary because maddrs shouldn't contain wonky things and we only have one query parameter as of this PR, but we should encode/decode URI component for this